tests/legacy-cli/e2e/tests/misc/browsers.ts

```diff
@@ -0,0 +1,63 @@
+import * as express from 'express';
+import * as path from 'path';
+import { copyProjectAsset } from '../../utils/assets';
+import { replaceInFile } from '../../utils/fs';
+import { ng } from '../../utils/process';
+
+export default async function () {
+  if (!process.env['E2E_BROWSERS']) {
+    return;
+  }
+
+  // Ensure SauceLabs configuration
+  if (!process.env['SAUCE_USERNAME'] || !process.env['SAUCE_ACCESS_KEY']) {
+    throw new Error('SauceLabs is not configured.');
+  }
+
+  await ng('build', '--prod');
+
+  // Add Protractor configuration
+  await copyProjectAsset('protractor-saucelabs.conf.js', 'e2e/protractor-saucelabs.conf.js');
+
+  // Remove browser log checks as they are only supported with the chrome webdriver
+  await replaceInFile(
+    'e2e/src/app.e2e-spec.ts',
+    'await browser.manage().logs().get(logging.Type.BROWSER)',
+    '[]',
+  );
+
+  // Workaround defect in getText WebDriver implementation for Safari/Edge
```

Comment:
Why not update the E2E schematics directly? I'm guessing the bug is also present if someone tries to run the E2E in Safari/Edge directly right?

Reply:
I felt that would be out of scope for this PR.

This is also a selenium/protractor deficiency which should really be fixed upstream.

The upcoming Protractor upgrade will also require a variety of changes for the E2E tests so I think it would be better to hold off on changes until then.